src/SDPoptimize.jl

```diff
@@ -9,207 +9,111 @@
 #############################################################################
 
 using ProgressMeter
+using Iterators
+using Interpolations
 
 """
-Convert the state and control tuples (stored as arrays) of the problem into integers
+Convert the state and control float tuples (stored as arrays or tuples) of the
+problem into integer tuples that can be used as indexes for the value function
 
 Parameters:
 - variable (Array)
     the vector variable we want to convert to an index (integer)
 
-- lower_bounds (Array)
+- bounds (Array)
     the lower bounds for each component of the variable
 
-- variable_sizes (Array)
-    the number of possibilities at each component knowing the upper and lower bounds
-    and the discretizations
-
 - variable_steps (Array)
     discretization step for each component
 
 
 Returns :
-- index (integer)
-    the index of the variable for the problem knowing the upper and lower bounds
-    as well as the discretization
+- index (tuple of integers)
+    the indexes of the variable
 
 """
-function index_from_variable( variable::Array,
-                    lower_bounds::Array,
-                    variable_sizes::Array,
+function index_from_variable( variable,
+                    bounds::Array,
                     variable_steps::Array)
-    index = 1;
-    j = 1;
-
-    for i = 1:length(variable)
-        index += j * (floor( Int, 1e-10 + ( variable[i] - lower_bounds[i] ) / variable_steps[i] ))
-        j *= variable_sizes[i]
-    end
 
-    return index
+    return tuple([floor(Int64, 1 + fld(1e-10+( variable[i] - bounds[i][1] ), variable_steps[i] )) for i in 1:length(variable)]...)
```

Comment:
Not so much, fld should return an Int but it returns here floats (which happened to be integers...) so I had to use floor(Int,64... Maybe I should keep using floor(y/x) instead of floor(fld(y,x))? I will benchmark this.